src/WebJobs.Script.Grpc/Extensions/ScriptInvocationContextExtensions.cs

```diff
@@ -30,21 +33,43 @@ public static async Task<InvocationRequest> ToRpcInvocationRequest(this ScriptIn
 
             foreach (var input in context.Inputs)
             {
-                TypedData rpcValue = null;
-                if (input.val == null || !rpcValueCache.TryGetValue(input.val, out rpcValue))
+                RpcSharedMemory sharedMem = null;
+                ParameterBinding parameterBinding = null;
+                if (isSharedMemoryDataTransferEnabled)
                 {
-                    rpcValue = await input.val.ToRpc(logger, capabilities);
-                    if (input.val != null)
+                    // Try to transfer this data over shared memory instead of RPC
+                    sharedMem = await input.val.ToRpcSharedMemoryAsync(logger, invocationRequest.InvocationId, sharedMemoryManager);
```

Comment:
Seems like all the changes to this method could be reverted if you just move the shared memory checks/handling into the existing ToRpc method where it belongs. Callers shouldn't have to do these checks.

Reply:
`ToRpc` always returns `TypedData` but we either want `RpcSharedMemory` if shared memory was used or `TypedData` if rpc was used.